### PR TITLE
Refactored and updated according to kubernetes 1.9 and openshift 3.9

### PR DESCRIPTION
**Pull request by piyush1594**

### CODE REVIEW

**Comment by piyush1594** [comment]
So all the client work has been done, now we need to release model to reflect here and all tests(unit and rt) passed on my local system.

@hrishin @oscerd @rohanKanojia @iocanel

**Comment by piyush1594** [comment]
@hrishin Upgraded with openshift 3.9.0 RT job and also latest released Kubernetes-Model 3.0.0 :tada:

**Comment by hrishin** [comment]
@piyush1594 @rohanKanojia what shall we do for OS 3.6, 3.7 RT's?

**Comment by hrishin** [comment]
@piyush1594 could you please update the comptability table?

**Comment by piyush1594** [comment]
Thanks @iocanel, will update the compatibility matrix and docs after CI get passes.

**Comment by piyush1594** [comment]
@hrishin Updated the compatibility matrix and also added docs regarding the major changes.

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/DeploymentIT.java [hunk 1]
```diff
@@ -57,54 +59,60 @@
...
       .build();
 
-    client.extensions().deployments().inNamespace(currentNamespace).createOrReplace(deployment1);
+    client.apps().deployments().inNamespace(currentNamespace).createOrReplace(deployment1);
   }
 
   @Test
   public void load() {
-    Deployment aDeployment = client.extensions().deployments().inNamespace(currentNamespace).load(getClass().getResourceAsStream("/test-deployments.yml")).get();
+    Deployment aDeployment = client.apps().deployments().inNamespace(currentNamespace).load(getClass().getResourceAsStream("/test-deployments.yml")).get();
```
Comment (hrishin) [inline kubernetes-itests/src/test/java/io/fabric8/kubernetes/DeploymentIT.java]: can we also ensure about client.extensions()? one use case is sufficient

File: kubernetes-itests/src/test/java/io/fabric8/kubernetes/NetworkPolicyIT.java [hunk 2]
```diff
@@ -74,13 +79,13 @@ public void init(){
       .endSpec()
       .build();
 
-    client.extensions().networkPolicies().create(networkPolicy);
+    client.network().networkPolicies().create(networkPolicy);
```
Comment (hrishin) [inline kubernetes-itests/src/test/java/io/fabric8/kubernetes/NetworkPolicyIT.java]: can we also ensure about client.extensions()? one use case is sufficient

**Comment by iocanel** [comment]
@piyush1594 @hrishin: This is a major blocker for us and I'd like to have this merged as soon as possible. Please let me know if there are things I can do to help with this PR.

Also, does anyone know why tests in CircleCI 3.6.0, 3.6.1, 3.7.0 take so long to run?

**Comment by oscerd** [comment]
What's the situation with 3.9 and 1.9 support?

**Comment by piyush1594** [comment]
@iocanel @oscerd @hrishin All checks passed :tada:

**Comment by iocanel** [comment]
Thanks @piyush1594 !!!

**Pull request merged**